src/client/client.c

```diff
@@ -57,6 +47,48 @@ int list_units_on(const char *name, Client *client) {
                 LIST_APPEND(units, unit_list->units, unit_ref(info));
         }
 
+        return r;
+}
+
+int list_units_on_all(Client *client) {
+        int r = 0;
+        _cleanup_unit_list_ UnitList *unit_list = new_unit_list();
+
+        r = asprintf(&client->object_path, "%s", HIRTE_OBJECT_PATH);
+        if (r < 0) {
+                return r;
+        }
+
+        r = fetch_unit_list(
+                        client,
+                        MANAGER_INTERFACE,
+                        NODE_AND_UNIT_INFO_STRUCT_TYPESTRING,
+                        &bus_parse_unit_on_node_info,
+                        unit_list);
+        if (r < 0) {
+                return r;
+        }
+
+        print_nodes_unit_list(&client->printer, unit_list);
+
+        return 0;
+}
+
+int list_units_on(const char *name, Client *client) {
+        int r = 0;
+        _cleanup_unit_list_ UnitList *unit_list = new_unit_list();
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, name, &client->object_path);
+        if (r < 0) {
+                return r;
```

Comment:
Should we `fprintf` an error message here?

Reply:
I don't think it's neccessary here. The only thing that can happen in this function is if we run out of memory, and it returns -ENOMEM.